### ISSUE TITLE
Option to not download ref files during singularity build

### ISSUE THREAD

**Issue by martinghunt** [iqbal-lab-org/clockwork#53]
The script `make_pipeline_reference_files.pl` is having issues downloading genomes (see #49 and #52).

It is run at the end of the singularity build:
https://github.com/iqbal-lab-org/clockwork/blob/master/singularity/clockwork_container.def#L29
It downloads TB reference genomes, and genomes used to check for contamination. It puts them inside the container, with the intention of maintaining easier reproducibility.

I propose we make the running of that script optional at singularity build time. It could be run post-build, and not put the ref files in the container. The advantages of downloading after building the container are:
1. that the script can continue where it left off, if it crashes. (note that it recognises genomes from the ENA and doesn't get them again, however it doesn't do that for hmpdacc, so ideally that could be fixed). Unlike during a singularity build, where any error unrecoverably aborts the build.
2. if clockwork isn't being used for Mtb, then none of these files are wanted anyway.

I have just had the script hang when running `enaDataGet`. I killed it and restarted and then it finished ok. Hence point 1.

**Comment by MartinGiap** [comment]
I find I can not connect to download tongue_dorsum.tar.bz2. I always get stuck with this file:

wget -nv -O Buccal_mucosa.tar.bz2 'ftp://public-ftp.hmpdacc.org/HMBSA/Buccal_mucosa.tar.bz2'
failed: Connection timed out.
failed: Connection timed out.
failed: Connection timed out.
failed: Connection timed out.
failed: Connection timed out.
failed: Connection timed out.
failed: Connection timed out.
failed: Connection timed out.
failed: Connection timed out.
failed: Connection timed out.
failed: Connection timed out.
failed: Connection timed out.
failed: Connection timed out.
failed: Connection timed out.
failed: Connection timed out.
failed: Connection timed out.
failed: Connection timed out.
failed: Connection timed out.
failed: Connection timed out.
failed: Connection timed out.
Error running: wget -nv -O Buccal_mucosa.tar.bz2 'ftp://public-ftp.hmpdacc.org/HMBSA/Buccal_mucosa.tar.bz2' at /clockwork/scripts/make_pipeline_reference_files.pl line 189.

How can I solve this problem?

**Comment by martinghunt** [comment]
@MartinGiap those files were moved, which broke the download. I just updated the download script with the new URLs -- see release 0.7.8 https://github.com/iqbal-lab-org/clockwork/releases/tag/v0.7.8.
Note that downloading is now separate from the build. You now need to build the container, then download the genomes ("set up reference genomes" section of https://github.com/iqbal-lab-org/clockwork/wiki/Pipelines has been updated).
